internal/output/vertical.go

```diff
@@ -0,0 +1,243 @@
+package output
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"unicode"
+
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/jedib0t/go-pretty/v6/text"
+)
+
+func PrintVerticalResults(vulnResult *models.VulnerabilityResults, outputWriter io.Writer) {
+	for i, result := range vulnResult.Results {
+		printVerticalHeader(result, outputWriter)
+		printVerticalVulnerabilities(result, outputWriter)
+
+		if len(vulnResult.ExperimentalAnalysisConfig.Licenses.Allowlist) > 0 {
+			printVerticalLicenseViolations(result, outputWriter)
+		}
+
+		if i < len(vulnResult.Results)-1 {
+			fmt.Fprintln(outputWriter)
+		}
+	}
+}
+
+func printVerticalHeader(result models.PackageSource, out io.Writer) {
+	fmt.Fprintf(
+		out,
+		"%s: found %s %s with issues\n",
+		text.FgMagenta.Sprintf("%s", result.Source.Path),
+		text.FgYellow.Sprintf("%d", len(result.Packages)),
+		Form(len(result.Packages), "package", "packages"),
+	)
+}
+
+func printVerticalVulnerabilitiesCountSummary(count int, state string, sourcePath string, out io.Writer) {
+	fmt.Fprintf(out, "\n  %s\n",
+		text.FgRed.Sprintf(
+			"%d %s %s found in %s",
+			count,
+			state,
+			Form(count, "vulnerability", "vulnerabilities"),
+			sourcePath,
+		),
+	)
+}
+
+func collectVulns(pkg models.PackageVulns, called bool) []models.Vulnerability {
+	vulns := make([]models.Vulnerability, 0)
+
+	for _, group := range pkg.Groups {
+		if group.IsCalled() != called {
+			continue
+		}
+
+		for _, id := range group.IDs {
```

Comment:
nit: it's a bit indented, maybe we can refactor this into a separate function.

Reply:
I feel like the awkward name of such a function would offset the gain in reducing indenting here